server/api/batch_inventory.py

```diff
@@ -209,6 +241,151 @@ def parse_vote(vote: str):
                 ),
             )
 
+    def process_ess():
+        ballot_count_by_batch: Dict[BatchKey, int] = defaultdict(int)
+        batch_tallies: Dict[BatchKey, Dict[str, int]] = defaultdict(
+            lambda: defaultdict(int)
+        )
+
+        def validate_choice_name_and_get_choice_id(choice_name: str) -> Optional[str]:
+            choice_id = None
+            for choice in contest.choices:
+                if choice.name == choice_name:
+                    choice_id = choice.id
+                    break
+
+            if (
+                not choice_id
+                and choice_name
+                and choice_name != "overvote"
+                and choice_name != "undervote"
+                and choice_name != "Write-in"
+            ):
+                raise UserError(f"Unrecognized choice in CVR file: {choice_name}")
+
+            return choice_id
+
+        # ZIP file with multiple CSVs
+        if batch_inventory_data.cvr_file.storage_path.endswith(".zip"):
+            entry_names = unzip_files(cvr_file, working_directory)
```

Comment:
This `unzip_files` function appears to have already been checked in, but I believe it to be unsafe. The documentation for [`zipfile`](https://docs.python.org/3/library/zipfile.html#resources-limitations) indicates that it is vulnerable to zip bombs. Here we extract every file in the `.zip` even though we're only interested in a few of them. The filtering done below should be done before extraction, for example.

I recognize that the risk is not high since you presumably have to have been logged in to be able to upload a malicious zip file and we control the sign up process. Still, I think we should change this and other uses of `unzip_files` to prevent this attack in the future.

Reply:
Wow nice find! Agree that we should update this. Opened an issue to track: https://github.com/votingworks/arlo/issues/1925